spec/lib/miq_expression/field_spec.rb

```diff
@@ -97,6 +97,12 @@
                                                               :associations => %w(win32_services),
                                                               :column       => 'dependencies')
     end
+
+    # This test is here because of the complexity of the MiqExpression::Field::REGEX.
+    # If the regex is removed or greatly simplified, then this test can also be removed.
+    it "avoids catastrophic backtracking" do
+      expect(Benchmark.realtime { MiqExpression::Field.match("XXXXXXXXXXXXXXXXXXXXXXXXX-") }).to be < 1.second
```

Comment:
Before this PR, this runs for a little under 7 seconds on my machine - I tried to pick a large enough string that even on a fast computer it would be longer than a second, but not so long that it makes the tests suite take forever.  I can throw another X in there which will bump the failure time from ~7 seconds to ~13 seconds (each X is about 2x time).